### PR TITLE
MGDSTRM-9567 MGDSTRM-9555 allowing for the transition from 3 to 2

### PR DESCRIPTION
**Pull request by shawkins**
also simplifying the logic based upon the possibility of dynamic scaling and node balancing - since we are enforcing balanced node scaling and have the topology spread constraint / scaling factor on brokers, we shouldn't need any deep node logic here.

### CODE REVIEW

**Comment by rareddy** [comment]
I guess this logic lets the replicas start at minimum of 1 node rather than minimum of 2.

**Comment by shawkins** [comment]
> I guess this logic lets the replicas start at minimum of 1 node rather than minimum of 2.

No, we haven't gotten sign off on such a change yet.

There's a check that enforces the extra instance - https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/pull/806/files#diff-89e55b846f9501475f8cd950173b4de15c56c06f2b9e53efee166862768000cdR502

File: operator/src/test/java/org/bf2/operator/managers/IngressControllerManagerTest.java [hunk 1]
```diff
@@ -121,6 +124,32 @@ public void testReplicaReduction() {
...
+            mk.getSpec().getCapacity().setEgressPerSec(Quantity.parse("300Mi"));
+            Kafka kafka = this.kafkaCluster.kafkaFrom(mk, null);
+            openShiftClient.resource(kafka).createOrReplace();
+        });
+        informerManager.createKafkaInformer();
+
+        ingressControllerManager.reconcileIngressControllers();
+        checkAzReplicaCount(3);
+
+        // remove two kafkas - we should keep the same number of replicas
```
Comment (grdryn) [inline operator/src/test/java/org/bf2/operator/managers/IngressControllerManagerTest.java:145]: Is this comment correct? My reading is that we expect 3 replicas on L143, but then 2 on L150, which I think contradicts the expectation of this comment?

Comment (shawkins) [reply operator/src/test/java/org/bf2/operator/managers/IngressControllerManagerTest.java:145]: Sorry, that is a cut and paste leftover.  Will remove.

Comment (grdryn) [reply operator/src/test/java/org/bf2/operator/managers/IngressControllerManagerTest.java]: Thanks!

**Review by grdryn — APPROVED** [review]
LGTM, thanks.

**Review by rareddy — APPROVED** [review]
lgtm

**Pull request merged**